### PR TITLE
ARROW-6121: [Tools] Improve merge tool ergonomics

### PR DESCRIPTION
**Pull request by fsaintjacques**
- merge_arrow_pr.py now accepts the pull-request number as a single optional argument, e.g. `./merge_arrow_pr.py 4921`.
- merge_arrow_pr.py can optionally read a configuration file located in   `~/.config/arrow/merge.conf` which contains options like jira credentials. See the `dev/merge.conf` file as example

### CODE REVIEW

**Review by kou — COMMENTED** [review]
I like this improvement.
How about renaming `dev/merge.conf` to `dev/merge.conf.example`, `dev/merge.conf.sample` or something? Developers may think that we can use `dev/merge.conf` directly without copying this to `~/.config.arrow/merge.conf`.

File: dev/merge_arrow_pr.py [hunk 1]
```diff
@@ -460,24 +461,48 @@ def get_version_json(version_str):
...
+    # Fallback to environment variables
+    if not username or not password:
+        username = os.environ.get("APACHE_JIRA_USERNAME")
+        password = os.environ.get("APACHE_JIRA_PASSWORD")
 
-    if not jira_username:
-        jira_username = cmd.prompt("Env APACHE_JIRA_USERNAME not set, "
-                                   "please enter your JIRA username:")
+    # Fallback to user tty prompt
+    if not username or not password:
```
Comment (pitrou) [inline dev/merge_arrow_pr.py]: Are we sure those are always set _both_? Should we not have:
```python
if not username:
    username = cmd.prompt(...)
if not password:
    password = cmd.getpass(...)
```

Comment (fsaintjacques) [reply dev/merge_arrow_pr.py]: That would be weird to set one and not the other, but we could.

Comment (pitrou) [reply dev/merge_arrow_pr.py]: Someone paranoid may not like to store a password in a conf file or an environment variable.

Comment (wesm) [reply dev/merge_arrow_pr.py]: We may as well make this change. I tested the script so we can merge this one this is addressed

**Review by wesm — APPROVED** [review]
+1

**Pull request closed without merging**